src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityConfiguration.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Class to store configuration for a managed identity enabled on a resource.
+    /// For a system assigned managed identity use ManagedIdentityConfiguration.SystemAssigned.
+    /// For user assigned managed identity use ManagedIdentityConfiguration.WithUserAssignedClientId("clientId") or 
+    /// ManagedIdentityConfiguration.WithUserAssignedResourceId("resourceId").
+    /// For more details see https://aka.ms/msal-net-managed-identity
+    /// </summary>
+    public class ManagedIdentityConfiguration
```

Comment:
Is the plan for this class to have any other non-Id related config also?

Since we have these static builders here, if we add more config to this class, I'd think we need to add more static builders for each config? That can get confusing...

My proposal:
- rename this class to `ManagedIdentityId`.
- It will still have the same logic - string ID, ID type, etc.
- It can also be added to MIAppOptions as a property - `public ManagedIdentityId ManagedIdentityId { get; set; }`
- Since it's a simple class, it can still be used in options when someone loads options from config or something.
- If there are more MI related config, we can add it to some other new class?

Reply:
My initial thought was also to add this to MIAppOptions as you mentioned above. I think the only concern here is the naming. 
Regarding the question, will this class be used to store anything other than id? I don't think so, it does not replace the MIAppBuilder. More config will be added there. The `WithUserAssignedClientId` / `WithUserAssignedResourceId` methods are helper methods to let the user create an instance of this class using client id or resource id. This class will be used to handle more Ids if needed, 

@bgavrilMS

Reply:
-  `WithUserAssignedClientId` `WithUserAssignedResourceId` should be removed. Let's not have 2 ways to configure the same thing, it leads to dev confusion. 
- We decided to remove `MIAppOptions` for now. It needs a bit more thought. We can add it later if we need it (pls open a issue).
- Naming is hard. `ManagedIdentityId` sounds ok to me.

Reply:
Discussed offline with @bgavrilMS, renamed to `ManagedIdentityId` and removed MIAppOptions